vmm/task/src/container.rs

```diff
@@ -245,28 +245,44 @@ impl KuasarFactory {
 }
 
 // runtime_error will read the OCI runtime logfile retrieving OCI runtime error
-pub async fn runtime_error(bundle: &str, e: runc::error::Error, msg: &str) -> Error {
+pub async fn runtime_error(bundle: &str, r_err: runc::error::Error, msg: &str) -> Error {
+    match get_last_runtime_error(bundle).await {
+        Err(e) => other!(
+            "{}: unable to retrieve OCI runtime error ({}): {}",
+            msg,
+            e,
+            r_err
+        ),
+        Ok(rt_msg) => {
+            if rt_msg.is_empty() {
+                other!("{}: empty msg in log file: {}", msg, r_err)
+            } else {
+                other!("{}: {}", msg, rt_msg)
+            }
+        }
+    }
+}
+
+async fn get_last_runtime_error(bundle: &str) -> Result<String> {
+    let log_path = Path::new(bundle).join("log.json");
     let mut rt_msg = String::new();
-    match File::open(Path::new(bundle).join("log.json")).await {
-        Err(err) => other!("{}: unable to open OCI runtime log file){}", msg, err),
+    // Use match because `await` is not allowed inside lambda
+    match File::open(log_path).await {
+        Err(e) => Err(other!("unable to open OCI runtime log file: {}", e)),
         Ok(file) => {
             let mut lines = BufReader::new(file).lines();
             while let Ok(Some(line)) = lines.next_line().await {
                 // Retrieve the last runtime error
                 match serde_json::from_str::<Log>(&line) {
-                    Err(err) => return other!("{}: unable to parse log msg: {}", msg, err),
+                    Err(e) => return Err(other!("unable to parse log msg: {}", e)),
```

Comment:
I suggest we can search for the most recent error-level log information in the log file by moving backward from the end of the file

Reply:
done